src/modules/functionality.js

```diff
@@ -0,0 +1,134 @@
+const form = document.getElementById('new-task-form');
+
+let taskStorage = [];
+let taskObject;
+
+const saveTodo = () => {
+  const allTodos = JSON.stringify(taskStorage);
+  localStorage.setItem('taskStorage', allTodos);
+};
+
+const getStoredTodos = () => {
+  taskStorage = JSON.parse(localStorage.getItem('taskStorage'));
+};
+
+const createTask = () => {
+  taskObject = {
+    description: input.value,
+    completed: false,
+    index: taskStorage.length + 1,
+  };
+  if (taskObject.description !== '') {
+    taskStorage.push(taskObject);
+    saveTodo();
+  }
+};
+
+export default function addEditRemoveTask(task) {
+  // function for adding new task to the list
+  const taskContainer = document.createElement('div');
+  taskContainer.classList.add('task');
+  const taskContentContainer = document.createElement('div');
+  taskContentContainer.classList.add('content');
+  taskContainer.appendChild(taskContentContainer);
+  const taskCheckboxContainer = document.createElement('input');
+  taskCheckboxContainer.classList.add('checkbox');
+  taskCheckboxContainer.type = 'checkbox';
+  const taskInputContainer = document.createElement('input');
+  taskInputContainer.classList.add('text');
+  taskInputContainer.type = 'text';
+  taskInputContainer.value = task.description;
+  taskInputContainer.setAttribute('readonly', 'readonly');
+  taskContentContainer.appendChild(taskCheckboxContainer);
+  taskContentContainer.appendChild(taskInputContainer);
+
+  // function for editing and removing task from the list
+  const taskActionsContainer = document.createElement('div');
+  taskActionsContainer.classList.add('actions');
+  const taskEditButton = document.createElement('button');
+  taskEditButton.classList.add('edit');
+  taskEditButton.innerText = 'Edit';
+  const taskDeleteButton = document.createElement('button');
+  taskDeleteButton.classList.add('delete');
+  taskDeleteButton.innerText = 'Delete';
+  taskActionsContainer.appendChild(taskEditButton);
+  taskActionsContainer.appendChild(taskDeleteButton);
+  taskContainer.appendChild(taskActionsContainer);
+  listContainer.appendChild(taskContainer);
+  input.value = '';
+
+  taskEditButton.addEventListener('click', () => {
+    if (taskEditButton.innerText.toLowerCase() === 'edit') {
+      taskEditButton.innerText = 'Save';
+      taskInputContainer.removeAttribute('readonly');
+      taskInputContainer.focus();
+    } else {
+      taskEditButton.innerText = 'Edit';
+      taskInputContainer.setAttribute('readonly', 'readonly');
+    }
+  });
+
+  // function for removing task from the list
+  taskDeleteButton.addEventListener('click', () => {
+    listContainer.removeChild(taskContainer);
+  });
+
+  // Remove the todo item from the list of todo items in local storage
+  taskDeleteButton.addEventListener('click', () => {
+    const taskIndex = taskStorage.indexOf(task);
+    taskStorage.splice(taskIndex, 1);
+    saveTodo();
+  });
```

Comment:
- [ ] Kindly make sure to update the task `index` of other remained tasks after removing the task. If the total task has for example 5 tasks and we remove the 3rd task make sure that the index of the task is updated in order to make sure that if the task is the fourth one must have an `index` of 4 every time.

Reply:
issue fixed. Thank you for your reviews